tests/driver/driver_test.py

```diff
@@ -235,8 +235,6 @@ def test_client_reaches_join_point_which_completes_parent(self):
         assert d.current_step == 0
         assert len(d.workers_completed_current_step) == 0
 
-        # this requires at least Python 3.6
-        # target.on_task_finished.assert_called_once()
```

Comment:
While we do have `assert_called_once()` now since we support Python 3.8+, asserting on `call_count` is fine and works for counts larger than 1, such as 4 on the line below.